### ISSUE TITLE
Why multiply mean by 0.0 in PPO1 mlp_policy?

### ISSUE THREAD

**Issue by sidmysore** [openai/baselines#446]
I was looking over the code in [mlp_policy.py](https://github.com/openai/baselines/blob/master/baselines/ppo1/mlp_policy.py) and I noticed this on [line 39](https://github.com/openai/baselines/blob/a6b1bc70f156dc45c0da49be8e80941a88021700/baselines/ppo1/mlp_policy.py#L39) (at least as of the current master):

`pdparam = tf.concat([mean, mean * 0.0 + logstd], axis=1)`

Could someone explain why the mean is multiplied by 0 before adding logstd? Unless I'm misunderstanding something, wouldn't it be equivalent to simply do this?:

`pdparam = tf.concat([mean, logstd], axis=1)`

Was the 0.0 perhaps meant to be a placeholder for something? The commit history seems to indicate that this line has been there since the file was created. Any clarification would be appreciated.

Thanks!

**Comment by AndiLi99** [comment]
As far as I can tell, mean is being concatenated along axis 1 and mean*0.0 returns an array of zeros that has the same shape as mean, in order to broadcast logstd into the correct shape.

**Comment by sidmysore** [comment]
I considered that, but I wasn't sure if that was the right interpretation.

If we look at [line 38](https://github.com/openai/baselines/blob/a6b1bc70f156dc45c0da49be8e80941a88021700/baselines/ppo1/mlp_policy.py#L38) though, where logstd is assigned:

`logstd = tf.get_variable(name="logstd", shape=[1, pdtype.param_shape()[0]//2], initializer=tf.zeros_initializer())`

and compare that to [line 37](https://github.com/openai/baselines/blob/a6b1bc70f156dc45c0da49be8e80941a88021700/baselines/ppo1/mlp_policy.py#L37), for mean:

`mean = tf.layers.dense(last_out, pdtype.param_shape()[0]//2, name='final', kernel_initializer=U.normc_initializer(0.01))`

it seems to me that they're already cast to the same shape. 

I could just be overthinking this whole thing, but I'm not sure, so I figured I'd seek clarification.

**Comment by initial-h** [comment]
same query in distributions.py
pdparam = tf.concat([mean, mean * 0.0 + logstd], axis=1)

**Comment by pzhokhov** [comment]
As @AndiLi99 said, the mean * 0.0 + logstd forces of broadcasting logstd  to the shape of mean. This is needed when mean is a tensorflow tensor with None components in the shape (i.e. batch size is undefined at graph construction time), and logstd is a variable of a fixed size. 
Consider the following snippet:
```
import tensorflow as tf
import numpy as np
mean = tf.placeholder(dtype=tf.float32, shape=[None, 10])
logstd = tf.Variable(np.zeros(10, 'float32'))

print(tf.concat([mean, mean * 0.0 + logstd], axis=-1).shape) # prints (?, 20)
print(tf.concat([mean, logstd], axis=-1).shape) # raises as exception
```
